### ISSUE TITLE
Echo/Noise in Quickblox audio call

### ISSUE THREAD

**Issue by kjp90** [QuickBlox/quickblox-javascript-sdk#410]
I have develop the videochat in Angular. I can refer Javascript SDK Sample. The sample is working fine. But when i developing the example in angular I get a lot of noise and echo in audio call. Video Call Working perfectly there is no noise or echo. Why is that so in audio call?

**Comment by austin92jus** [comment]
can you send your snapshot of the code (the function of getUserMedia) then only we can get to know what you are missed? 

thank you

**Comment by austin92jus** [comment]
var mediaParams = {
  audio: true,
  video: true,
  options: {
    muted: true,
    mirror: true,
  },
  elemId: "localVideo",
};

you need to mention muted as a true
else it will lead to echoing your voice ..

Noise webrtc defaulty suppress the noise

**Comment by kjp90** [comment]
Tahnk you for your reply. 

 private getMediaParams(isAudio: boolean, localVideo: string): Object{
        let mediaParams = {};
        console.log("ISAUDIO--->",isAudio)
        if(isAudio){
            mediaParams = {
                audio: true,
                video: false,
                options: {
                    muted: true,
                    // mirror: true
                }
            };
        } else {
            mediaParams = {
                audio: true,
                video: true,
                elemId: localVideo,
                options: {
                    muted: true,
                    mirror: true
                }
            };
        }

        return mediaParams;
    }

**Comment by austin92jus** [comment]
i think you are facing this issues only on audio call?

**Comment by kjp90** [comment]
yes. Video call is working perfectly. no issues. audio call only having the echo & noise issue

**Comment by austin92jus** [comment]
mediaParams = {
audio: true,
video: true,
elemId: localVideo,
options: {
muted: true,
}

you also need to mentioned elemId for audio call as well

in order to minimized the echoing we need to attach the stream to one element then only we can muthed the stream


Or else you can do another way 
get the stream from getUsermedia 
and mute the stream object

**Comment by austin92jus** [comment]
mediaParams = {
audio: true,
video: true,
elemId: localVideo,
options: {
muted: true,
}

you also need to mentioned elemId for audio call as well

in order to minimized the echoing we need to attach the stream to one element then only we can muthed the stream


Or else you can do another way 
get the stream from getUsermedia 
and mute the stream object

**Comment by kjp90** [comment]
ok . Thank You. I will try this.

**Comment by kjp90** [comment]
Hai I was tried this but it also not working. any other solutions. You have any code for this.

**Comment by ghost** [comment]
Hi @kjp90 ,
You can pass [MediaTrackConstraints](https://developer.mozilla.org/en-US/docs/Web/API/MediaTrackConstraints) for `audio` and / or `video` key(s)
```javascript
const mediaParams = {
  audio: {
    echoCancellation: true,
    noiseSuppression: true
  },
  video: false,
  options: { muted: true }
}
```
